# Shopping Mall Customer Data Segmentation Analysis

## Project Objective

The objective of this project is to analyze customer demographics, annual income, and spending behavior to identify different customer segments. The analysis aims to generate actionable business insights that can improve customer targeting, retention, and overall sales performance.

## Part 1: Data Cleaning & Understanding

In [2]:
import pandas as pd

In [3]:
df = pd.read_csv(r"C:\Users\SRIKANT SENAPATI\Downloads\Shopping Mall Customer Segmentation Data .csv")

In [4]:
df.head()

,Customer ID,Age,Gender,Annual Income,Spending Score
0,d410ea53-6661-42a9-ad3a-f554b05fd2a7,30,Male,151479,89
1,1770b26f-493f-46b6-837f-4237fb5a314e,58,Female,185088,95
2,e81aa8eb-1767-4b77-87ce-1620dc732c5e,62,Female,70912,76
3,9795712a-ad19-47bf-8886-4f997d6046e3,23,Male,55460,57
4,64139426-2226-4cd6-bf09-91bce4b4db5e,24,Male,153752,76


In [5]:
df.columns = df.columns.str.lower().str.replace(' ','_').str.strip()

In [6]:
df.head()

,customer_id,age,gender,annual_income,spending_score
0,d410ea53-6661-42a9-ad3a-f554b05fd2a7,30,Male,151479,89
1,1770b26f-493f-46b6-837f-4237fb5a314e,58,Female,185088,95
2,e81aa8eb-1767-4b77-87ce-1620dc732c5e,62,Female,70912,76
3,9795712a-ad19-47bf-8886-4f997d6046e3,23,Male,55460,57
4,64139426-2226-4cd6-bf09-91bce4b4db5e,24,Male,153752,76


In [7]:
df.shape

(15079, 5)

In [8]:
df['gender'].unique()

array(['Male', 'Female'], dtype=object)

In [9]:
df.isnull().sum()

customer_id       0
age               0
gender            0
annual_income     0
spending_score    0
dtype: int64

In [10]:
df.duplicated().sum()

np.int64(0)

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15079 entries, 0 to 15078
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   customer_id     15079 non-null  object
 1   age             15079 non-null  int64 
 2   gender          15079 non-null  object
 3   annual_income   15079 non-null  int64 
 4   spending_score  15079 non-null  int64 
dtypes: int64(3), object(2)
memory usage: 589.2+ KB


## Part 2: Customer Analysis

#### Q1. What is the Average Age of Customers?

In [12]:
df['age'].mean()

np.float64(54.191590954307316)

### Insight

The average age of customers is approximately 54 years, indicating that the customer base is dominated by middle-aged and senior individuals.

#### Q2. What is the Average Annual Income ?

In [13]:
df['annual_income'].mean()

np.float64(109742.88056237152)

### Insight

The average annual income is approximately 109,743. This suggests that customers belong to a wide range of income categories.

#### Q3. Which Gender Spends More on Average?

In [14]:
df.groupby('gender')['spending_score'].mean().reset_index()

,gender,spending_score
0,Female,50.397648
1,Male,50.782752


### Insight

Male customers have a slightly higher average spending score than female customers. However, the difference is very small, indicating similar spending behavior across genders.

#### Q4. Which Age Group Has the Highest Spending Score?

In [15]:
bins = [0,20,30,40,50,60,100]

labels = ['0-20','21-30','31-40','41-50','51-60','60+']

df['age_group'] = pd.cut(df['age'],bins=bins,labels=labels)

df.groupby('age_group')['spending_score'].mean().reset_index()

C:\Users\SRIKANT SENAPATI\AppData\Local\Temp\ipykernel_9848\207430031.py:7: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  df.groupby('age_group')['spending_score'].mean().reset_index()


,age_group,spending_score
0,0-20,51.397727
1,21-30,50.845057
2,31-40,50.545894
3,41-50,50.564746
4,51-60,51.272989
5,60+,50.225445


### Insight

The average spending score is nearly identical across all age groups (50.27–51.40), indicating that age has little influence on customer spending behavior in this dataset. Since no age group shows significantly higher or lower spending, age alone is not a strong factor for customer segmentation or targeted marketing.

#### Q5. Is There Any Relationship Between Annual Income and Spending Score?

In [16]:
df['annual_income'].corr(df['spending_score'])

np.float64(0.0032341193354843457)

### Insight

The correlation coefficient is very close to zero, indicating almost no relationship between annual income and spending score. Income alone does not explain customer spending behavior.

## Part 3: Customer Segmentation

#### Creating Customer Segments

In [17]:
income_threshold = df['annual_income'].median()

spending_threshold = df['spending_score'].median()

def segment(row):

    if row['annual_income'] >= income_threshold and row['spending_score'] >= spending_threshold:
        return "High Income - High Spending"

    elif row['annual_income'] >= income_threshold and row['spending_score'] < spending_threshold:
        return "High Income - Low Spending"

    elif row['annual_income'] < income_threshold and row['spending_score'] >= spending_threshold:
        return "Low Income - High Spending"

    else:
        return "Low Income - Low Spending"

df['segment'] = df.apply(segment, axis=1)

In [18]:
df.head()

,customer_id,age,gender,annual_income,spending_score,age_group,segment
0,d410ea53-6661-42a9-ad3a-f554b05fd2a7,30,Male,151479,89,21-30,High Income - High Spending
1,1770b26f-493f-46b6-837f-4237fb5a314e,58,Female,185088,95,51-60,High Income - High Spending
2,e81aa8eb-1767-4b77-87ce-1620dc732c5e,62,Female,70912,76,60+,Low Income - High Spending
3,9795712a-ad19-47bf-8886-4f997d6046e3,23,Male,55460,57,21-30,Low Income - High Spending
4,64139426-2226-4cd6-bf09-91bce4b4db5e,24,Male,153752,76,21-30,High Income - High Spending


#### Segment Distribution

### Insight

The customer base can be divided into four meaningful groups based on income and spending behavior. These segments help businesses design targeted marketing strategies and improve customer engagement.

In [19]:
df.to_csv(r'C:\Users\SRIKANT SENAPATI\Downloads\Cleaned_Shopping Mall Customer Segmentation Data.csv',index=False)

In [20]:
df.head()

,customer_id,age,gender,annual_income,spending_score,age_group,segment
0,d410ea53-6661-42a9-ad3a-f554b05fd2a7,30,Male,151479,89,21-30,High Income - High Spending
1,1770b26f-493f-46b6-837f-4237fb5a314e,58,Female,185088,95,51-60,High Income - High Spending
2,e81aa8eb-1767-4b77-87ce-1620dc732c5e,62,Female,70912,76,60+,Low Income - High Spending
3,9795712a-ad19-47bf-8886-4f997d6046e3,23,Male,55460,57,21-30,Low Income - High Spending
4,64139426-2226-4cd6-bf09-91bce4b4db5e,24,Male,153752,76,21-30,High Income - High Spending
